packages/react/src/components/table/DataTable.tsx

```diff
@@ -0,0 +1,266 @@
+import React, { useEffect, useImperativeHandle, useMemo, useState } from 'react';
+
+// import core base styles
+import 'hds-core';
+import styles from './Table.module.scss';
+import { Table } from './Table';
+import { Checkbox } from '../checkbox';
+import { useTheme } from '../../hooks/useTheme';
+
+type Header = {
+  key: string;
+  headerName: string;
+  sortIconType?: 'string' | 'other';
+  transform?: ({ args }: any) => string | JSX.Element; // eslint-disable-line @typescript-eslint/no-explicit-any
+};
+
+export interface TableCustomTheme {
+  '--background-color'?: string;
+}
+
+export type DataTableProps = React.ComponentPropsWithoutRef<'table'> & {
+  cellConfig: {
+    cols: Array<Header>;
+    sortingEnabled?: boolean;
+    initialSortingColumnKey?: string; // undefined -> neutral order for all columns
+    initialSortingOrder?: 'asc' | 'desc';
+    ariaLabelSortButtonNeutral?: string;
+    ariaLabelSortButtonAscending?: string;
+    ariaLabelSortButtonDescending?: string;
+    indexKey: string; // column key used as unique identifier for row
+    renderIndexCol?: boolean; // whether index colum is rendered in table. Defaults to true.
+  };
+  rows: Array<object>;
+  verticalHeaders?: Array<Header>;
+  variant?: 'dark' | 'light';
+  dense?: boolean;
+  zebra?: boolean;
+  caption?: string | React.ReactNode;
+  verticalLines?: boolean;
+  textAlignContentRight?: boolean; // defaults to false -> text is aligned left
+  checkboxSelection?: boolean;
+  ariaLabelCheckboxSelection?: string;
+  setSelections?: Function; // Callback that gets called with all selected row id values
+  initiallySelectedRows?: any[]; // Initially selected rows. Apply corresponding indexKey values here.
+  /**
+   * Custom theme styles
+   */
+  theme?: TableCustomTheme;
+};
+
+function processRows(rows, order, sorting, cellConfig) {
+  if (!cellConfig.sortingEnabled || !order || !sorting) {
+    return [...rows];
+  }
+
+  const sortColumn = cellConfig.cols.find((column) => {
+    return column.key === sorting;
+  });
+
+  const customSortCompareFunction = sortColumn ? sortColumn.customSortCompareFunction : undefined;
+
+  if (customSortCompareFunction) {
+    const sortedRows = [...rows].sort((a, b) => {
+      const aValue = a[sorting];
+      const bValue = b[sorting];
+
+      return customSortCompareFunction(aValue, bValue);
+    });
+
+    if (order === 'asc') {
+      return sortedRows;
+    }
+    if (order === 'desc') {
+      return sortedRows.reverse();
+    }
+  }
+
+  return [...rows].sort((a, b) => {
+    const aValue = a[sorting];
+    const bValue = b[sorting];
+
+    if (aValue < bValue) {
+      return order === 'asc' ? -1 : 1;
+    }
+    if (aValue > bValue) {
+      return order === 'asc' ? 1 : -1;
+    }
+
+    return 0;
+  });
+}
+
+export const DataTable = React.forwardRef(
```

Comment:
Should we change this name to be just 'Table'? I named it to DataTable because there was already a Table component that it applies. If we wish to name this to just 'Table', we need to figure a new name to the Table component that it uses.

Reply:
This has been renamed as 'Table'